Marlin/Marlin_main.cpp

```diff
@@ -1579,7 +1647,7 @@ static void setup_for_endstop_move() {
       feedrate = homing_feedrate[Z_AXIS];
 
       // Move down until the Z probe (or endstop?) is triggered
-      float zPosition = -(Z_MAX_LENGTH + 10);
+      float zPosition = MIN_PROBE_PT;
```

Comment:
@thinkyhead I think I see what you mean, when one point successfully probes the current position may not be accurate after the pin is toggled. From our tests this works fine.

In fact, we've even published a commit in our repo that sets the position of an axis when an endstop (not the reference one used for homing) is hit to `(float)endstops_trigsteps[_AXIS(AXIS)]/axis_steps_per_unit[_AXIS(AXIS)]`. That way, when an endstop is hit the position is kept to the current position on that axis. Take a loot at https://github.com/alephobjects/Marlin/commit/ec71edafb2ac42f42d6dc4926f4a3684452df980 to see what I'm talking about.